packages/ember-data/lib/system/model/model.js

```diff
@@ -901,4 +902,19 @@ Model.reopenClass({
   modelName: null
 });
 
+if (Ember.platform.hasPropertyAccessors) {
+  var errorDeprecationShown = false;
+
+  Ember.defineProperty(Model.prototype, 'error', {
+    enumerable: false,
+    get: function() {
+      if (!errorDeprecationShown) {
+        Ember.deprecate('DS.Model#error has been deprecated please use adapterError instead');
+        errorDeprecationShown = true;
+      }
+      return Ember.get(this, 'adapterError');
+    }
+  });
+}
```

Comment:
Lets just make this a normal computed property so it can be overridden. Part of the original reason for renaming error to `adapterError` was because people were defining an `error` attribute on their models. That should still be allowed.